internal/engine/upper_test.go

```diff
@@ -547,10 +548,7 @@ func TestNoOpChangeToDockerfile(t *testing.T) {
 	// incremental build.
 	call = <-f.b.calls
 	assert.Equal(t, "foobar", string(call.manifest.Name))
-	assert.ElementsMatch(t, []string{
-		f.JoinPath("Dockerfile"),
-		f.JoinPath("random_file.go"),
-	}, call.state.FilesChanged())
+	assert.ElementsMatch(t, []string{f.JoinPath("random_file.go")}, call.state.FilesChanged())
```

Comment:
so the result of my magic in upper.go is that if config files change but don't result in a manifest change, i remove config files from `state.FilesChanged()` (so that it just represents files we want to copy/rm from the container). idk if this is shifty -- thoughts?

Reply:
...tbh i don't understand why in the old code we don't freak out trying to process `Tiltfile` as a changed file -- but before i modified `ms.PendingFileChanges` builds were failing with:
```
build failed: file /Users/.../servantes/Tiltfile matches no mounts
```
maybe a difference in how we process changed files in image build vs container build? idk...